### PR TITLE
Added Firebreather trainer class

### PR DESCRIPTION
**Pull request by Alekemon**
As a test to how this engine works I wanted to start by adding a trainer class

### CODE REVIEW

**Comment by chacolah** [comment]
Have you tested it at all?

**Comment by Alekemon** [comment]
> Have you tested it at all?

I tried, I checked all errors marked by npm and fixed them but I can't get to open the localhost page

**Comment by Alekemon** [comment]
Ok I opened inspect and notified another error, going to fix it

**Comment by Alekemon** [comment]
Now the game opens in localhost:8000, I'll test when a firebreather battle pops in

**Comment by Alekemon** [comment]
Is there a way to force a trainer battle to ocur in debug to save time?

**Comment by Alekemon** [comment]
Now it works as intended

**Comment by TempsRay** [comment]
Seems okay to me. No dialogue though?

**Comment by Alekemon** [comment]
I've implemented dialogue now

File: src/data/dialogue.ts [hunk 1]
```diff
@@ -414,7 +414,17 @@ export const trainerTypeDialogue = {
         `Time to rev up and leave you in the dust!`
       ],
       victory: [
-        `I'll tune up for the next race.`
+        `If we only had it warmer.`
+      ]
+    }
+  ],,
```
Comment (LaukkaE) [inline src/data/dialogue.ts:420]: Double colon here

Comment (Alekemon) [reply src/data/dialogue.ts]: Thanks for notifying the oversight! I fixed that

**Comment by LaukkaE** [comment]
Trainer spawns correctly in Volcano, and seems to work right. 
Could use more dialogue if available.

**Comment by Alekemon** [comment]
> Trainer spawns correctly in Volcano, and seems to work right. Could use more dialogue if available.

I'll take a look at dialogues from Pokémon HGSS (where the Firebreather trainer class originates from) to implement into the trainer class

**Comment by Alekemon** [comment]
I've added some more dialogue lines

**Review by LaukkaE — APPROVED** [review]

File: src/data/dialogue.ts [hunk 2]
```diff
@@ -418,6 +418,20 @@ export const trainerTypeDialogue = {
...
+    {
+      encounter: [
+        `My flames shall devour you!`,
+        `I'm seriously upset by your cold confusion!`,
+        `I'm going to put on my fire-breathing shows!`
+      ],
+      victory: [
+        `If we only had it warmer.`,
+        `I flamed out.`
+
```
Comment (molker) [inline src/data/dialogue.ts:431]: can we get one more victory dialogue to balance out the encounter and victory count?

Comment (Alekemon) [reply src/data/dialogue.ts:432]: Sure, I added a 3rd dialogue now

**Pull request closed without merging**